auth/simple_token.go

```diff
@@ -52,6 +54,21 @@ func (as *authStore) assignSimpleTokenToUser(username, token string) {
 	}
 
 	as.simpleTokens[token] = username
+	timer := time.NewTimer(SimpleTokenTTL)
+	donec := make(chan struct{})
+	as.simpleTokenInf[token] = simpleTokenInfo{tokenTimer: timer, done: donec}
+	go func() {
```

Comment:
we ourselves iterating and comparing the deadlines with the current time seems like we  ourselves implementing a Timer functionality.

perhaps better idea is to use 

```
func AfterFunc(d Duration, f func()) *Timer
AfterFunc waits for the duration to elapse and then calls f in its own goroutine. It returns a Timer that can be used to cancel the call using its Stop method.

```
the function passed can delete the token. what do you say.